src/traits.rs

```diff
@@ -102,6 +102,15 @@ pub trait Object: Sized {
     /// Error type returned by handler methods
     type Error;
 
+    /// Defines how many of this type of object should be cached
+    fn cache_max_capacity() -> u64 {
+        1000
+    }
+    /// Defines how long objects of this type should live in the in-memory cache
+    fn cache_time_to_live() -> Duration {
+        Duration::from_secs(600)
+    }
```

Comment:
Makes more sense to move this to the FederationConfig. Should also be disabled in debug mode imo.

Reply:
I actually did this intentionally in the trait so the configuration can depend on the object. For example, it makes sense to cache a lot of communities and posts and cache them for a long time while caching Vote objects is probably not very useful.

Reply:
Votes dont implement the Object trait so they wont get cached either way.

Reply:
Ah, ok. But still, does it make sense what I wrote? It probably makes sense to have pretty much all communities in the cache but only the most recent 1000 posts or so, while caching comments potentially is not too useful (since those are mostly queried in batches or together with a post)

I agree that it should probably also somehow be configurable by user though, or somehow dependent on system memory.